### PR TITLE
Small Wifi module optimization and improvements.

### PR DESCRIPTION
**Pull request by MatthewvanBuuren**
Add Wifi status to Heartbeat. Updating WIFI_ENABLED usage to remove unrequired dependent capabilities.

## Description
- Add Wifi Status to Heartbeat; Useful for my use-case where various issues cause loss of access and this helps diagnose.
- Optimized pre-processing of network.h/.cpp when WIFI_ENABLED is disabled.
- Add  WIFI_ENABLED checks where required/necessary on transitive dependencies. i.e. #if ENABLE_WIFI && INCOMING_WIFI_ENABLED

- Samples from WIFI_ENABLED|AUDIO_ENABLED * 0|1 against single env.
----
audio, wifi
(I) (loop)(C1) Wifi: WL_CONNECTED, IP: 10.0.0.41, Mem: 46088, LargestBlk: 40948, PSRAM Free: 0/0, Buffer: 46/7, LED FPS: 0, LED Watt: 1079574528, LED Brite: 0%, Audio FPS: 0, Serial FPS: 1086324736, PeakVU: 0.00, MinVU: 1.00, VURatio: 1.20, CPU: 06%, 00%, FreeDraw: -1.407352

----
audio, no wifi
(I) (loop)(C1) Mem: 77584, LargestBlk: 59380, PSRAM Free: 0/0, Buffer: 0/369, LED FPS: 46, LED Watt: 7, LED Brite: 100%, Audio FPS: 35, Serial FPS: 0, PeakVU: 0.00, MinVU: 0.00, VURatio: 0.00, CPU: 00%, 19%, FreeDraw: 0.018975

----
wifi, no audio
(I) (loop)(C1) Wifi: WL_CONNECTION_LOST, IP: 0.0.0.0, Mem: 45916, LargestBlk: 42996, PSRAM Free: 0/0, Buffer: 0/286, LED FPS: 46, LED Watt: 8, LED Brite: 100%, CPU: 02%, 34%, FreeDraw: 0.016148

----
default
(I) (loop)(C1) Mem: 88064, LargestBlk: 61428, PSRAM Free: 0/0, Buffer: 0/375, LED FPS: 45, LED Watt: 7, LED Brite: 100%, CPU: 00%, 05%, FreeDraw: 0.015849

## Contributing requirements
* [x] I read the contribution guidelines in [CONTRIBUTING.md](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/CONTRIBUTING.md).
* [x] I understand the BlinkenPerBit metric, and maximized it in this PR.
* [x] I selected `main` as the target branch.
* [x] All code herein is subjected to the license terms in [COPYING.txt](http://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/main/COPYING.txt).

### CODE REVIEW

**Comment by MatthewvanBuuren** [comment]
Previous PR for context: https://github.com/PlummersSoftwareLLC/NightDriverStrip/pull/175

File: src/network.cpp [hunk 1]
```diff
@@ -150,18 +152,18 @@ void IRAM_ATTR RemoteLoopEntry(void *)
...
 bool ConnectToWiFi(uint cRetries)
 {
-    #if !ENABLE_WIFI
-        return false;
-    #endif
-
+    // Already connected, Go no further.
+    if (WiFi.isConnected())
+    {
+            return true;
```
Comment (rbergen) [inline src/network.cpp:164]: This line seems to be indented too much respective to the code around it.

File: src/main.cpp [hunk 2]
```diff
@@ -855,14 +859,22 @@ void loop()
                 debugI("Mem: %u LED FPS: %d, LED Watt: %d, LED Brite: %0.0lf%%, Audio FPS: %d, Serial FPS: %d, PeakVU: %0.2lf, MinVU: %0.2lf, VURatio: %0.2lf, CPU: %02.0f%%, %02.0f%%, FreeDraw: %lf",
                         ESP.getFreeHeap(),
                         g_FPS, g_Watts, g_Brite, g_AudioFPS, g_serialFPS, gPeakVU, gMinVU, gVURatio, g_TaskManager.GetCPUUsagePercent(0), g_TaskManager.GetCPUUsagePercent(1), g_FreeDrawTime);
-            #else
-            debugI("IP: %s, Mem: %u LargestBlk: %u PSRAM Free: %u/%u Buffer: %d/%d LED FPS: %d, LED Watt: %d, LED Brite: %0.0lf%%, CPU: %02.0f%%, %02.0f%%, FreeDraw: %lf",
+            #elif ENABLE_WIFI
```
Comment (rbergen) [inline src/main.cpp:862]: I wonder if it makes sense to also include (that is, start with) a debug variation for when `ENABLE_AUDIO` and `ENABLE_WIFI` are both set?

File: src/network.cpp [hunk 3]
```diff
@@ -150,18 +152,18 @@ void IRAM_ATTR RemoteLoopEntry(void *)
 // ConnectToWiFi
 //
 // Connect to the pre-configured WiFi network.  
-//
-// BUGBUG I'm guessing this is exposed in all builds so anyone can call it and it just returns false if wifi
-// isn't being used, but do we need that?  If no one really needs to call it put the whole thing in the ifdef
 
+#if ENABLE_WIFI
```
Comment (rbergen) [inline src/network.cpp:156]: As before, I think it's clearer if the code between this `#if ENABLE_WIFI` directive and its corresponding `#endif` is indented.

**Review by rbergen — COMMENTED** [review]
In general. I think that code between `#if`/`#endif` pairs you introduce should be indented consistent with the code that's already in place. A few of my in-line comments concern places where the lack of indentation is particularly confusing.

You can confirm that you have successfully built the version with the updates in this PR with `ENABLE_WIFI` both set and unset?

File: src/network.cpp (hunk 1 above)
Comment (MatthewvanBuuren) [reply src/network.cpp:164]: Good catch.

File: include/network.h [hunk 4]
```diff
@@ -31,11 +31,12 @@
 #include "ntptimeclient.h"
 #include "secrets.h"                          // copy include/secrets.example.h to include/secrets.h
 
+#if ENABLE_WIFI
```
[the remark this answers is not in the document]
Comment (MatthewvanBuuren) [reply include/network.h:34]: ACK, Noted.

**Comment by rbergen** [comment]
CI fails for the chieftain env with the following error: 

```text
In file included from src/main.cpp:165:
include/network.h:64:27: error: redefinition of 'const char* WLtoString(wl_status_t)'
 inline static const char* WLtoString(wl_status_t status) {
Compiling .pio/build/chieftain/src/network.cpp.o
                           ^~~~~~~~~~
In file included from include/soundanalyzer.h:45,
                 from src/main.cpp:163:
include/network.h:64:27: note: 'const char* WLtoString(wl_status_t)' previously defined here
 inline static const char* WLtoString(wl_status_t status) {
                           ^~~~~~~~~~
*** [.pio/build/chieftain/src/main.cpp.o] Error 1
========================= [FAILED] Took 22.95 seconds =========================

Environment    Status    Duration
-------------  --------  ------------
chieftain      FAILED    00:00:22.952
==================== 1 failed, 0 succeeded in 00:00:22.952 ====================
```

The problem seems to be caused by `network.h` being included multiple times, leading to multiple definitions for the inline function you've added to that file.

File: src/main.cpp (hunk 2 above)
Comment (MatthewvanBuuren) [reply src/main.cpp:862]: OFC yep.

**Comment by rbergen** [comment]
Just noting that adding a `#pragma once` just above the first `#include` in `network.h` solves the problem with the chieftain env build on my local machine.

**Comment by MatthewvanBuuren** [comment]
> Just noting that adding a `#pragma once` just above the first `#include` in `network.h` solves the problem with the chieftain env build on my local machine.

Is there a built-target for running the gauntlet? For next time.

File: include/network.h [hunk 5]
```diff
@@ -25,23 +25,55 @@
...
-#if INCOMING_WIFI_ENABLED
-    extern SocketServer g_SocketServer;
-#endif
-void processRemoteDebugCmd();
+#if ENABLE_WIFI
+    extern uint8_t g_Brightness;
+    extern bool g_bUpdateStarted;
+    extern WiFiUDP g_Udp;
+    #if INCOMING_WIFI_ENABLED
+        extern SocketServer g_SocketServer;
```
Comment (MatthewvanBuuren) [inline include/network.h:40]: Taking a look around the code-base there is a mix of indent includes/extern, all other blocks seem to follow indent so went with that.

**Comment by rbergen** [comment]
FYI, because we're touching a rather important part of the overall project, I've asked Dave if it would be possible for him to test the changes (with WiFi enabled and disabled) on his actual hardware. I'll keep you updated on the developments concerning this.

**Comment by rbergen** [comment]
Merging this based on absence of objections from Dave.

**Pull request merged**